src/gorilla.h

```diff
@@ -0,0 +1,63 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#ifndef GORILLA_H
+#define GORILLA_H
+
+#include <sys/types.h>      // u_int_t
+#include <stdbool.h>        // bool
+
+#define timestamp_t u_int64_t
+
+typedef union {
+    double d;
+    int64_t i;
+    u_int64_t u;
+} union64bits;
+
+typedef struct CompressedChunk {
+    u_int64_t size;
+    u_int64_t count;
+    u_int64_t idx;
+
+    union64bits baseValue;
+    u_int64_t baseTimestamp;
+
+    u_int64_t *data;
+    
+    u_int64_t prevTimestamp;
+    int64_t prevTimestampDelta;
+
+    union64bits prevValue;
+    u_int8_t prevLeading;
+    u_int8_t prevTrailing;
+} CompressedChunk;
+
+typedef struct CChunk_Iterator {
+  CompressedChunk *chunk;
+  u_int64_t idx;
+  u_int64_t count;
+
+  // timestamp vars
+  u_int64_t prevTS;
+  int64_t prevDelta;
+
+  // value vars
+  union64bits prevValue;  
+  u_int8_t prevLeading;
+  u_int8_t prevTrailing;
+} CChunk_Iterator;
+
+enum result {
```

Comment:
Uppercase Result. Also, since we don't have namespaces, it might get in conflict with other headers, so we may want to add some prefix.

Reply:
Move it to const.h and rename enum to Chunk_Result.

Reply:
> Move it to const.h and rename enum to Chunk_Result.

Done in commit https://github.com/RedisTimeSeries/RedisTimeSeries/pull/261/commits/383d425766245639d42c4756dcd111faf38c6d61